code/modules/mob/living/basic/drone/extra_drone_types.dm

```diff
@@ -71,27 +47,48 @@
 	title = ROLE_SYNDICATE_DRONE
 	policy_index = ROLE_SYNDICATE_DRONE
 
+/// A version of the syndrone that gets a nuclear uplink, a firearms implant, and 30 TC.
+/mob/living/basic/drone/syndrone/badass
+	name = "Badass Syndrone"
+	default_storage = /obj/item/uplink/nuclear
+
+/mob/living/basic/drone/syndrone/badass/Initialize(mapload)
+	. = ..()
+	var/datum/component/uplink/hidden_uplink = internal_storage.GetComponent(/datum/component/uplink)
+	hidden_uplink.set_telecrystals(30)
```

Comment:
Any reason this isn't a var on the type to avoid using `GetComponent`?

Reply:
I'm actually not sure how we'd get the component into a var in the first place without using GetComponent, since the drone uplink is just the default uplink radio.

Reply:
No I mean that on `/mob/living/basic/drone/syndrone` you create the component and set its telecrystals to 10 and then in this one you get the component added in super and set the telecrystals to 30 instead.
Why not just make the telecrystals to add into a var rather than setting it twice?

Reply:
Oh! Right. I understand now.